stdlib/list.mli

```diff
@@ -102,6 +102,39 @@ val flatten : 'a list list -> 'a list
 (** An alias for [concat]. *)
 
 
+(** {1 Comparison} *)
+
+val equal : ('a -> 'a -> bool) -> 'a list -> 'a list -> bool
+(** [equal eq [a1; ...; an] [b1; ..; bm]] holds when
+    the two input lists have the same length, and for each
+    pair of elements [ai], [bi] at the same position we have
+    [eq ai bi].
+
+    Note: the [eq] function may be called even if the
+    lists have different length. If you know your equality
+    function is costly, you may want to check {!compare_lengths}
+    first.
+
+    @since 4.12.0
+*)
+
+val compare : ('a -> 'a -> int) -> 'a list -> 'a list -> int
+(** [compare cmp [a1; ...; an] [b1; ...; bm]] performs
+    a lexicographic comparison of the two input lists,
+    using the same ['a -> 'a -> int] interface as {!Stdlib.compare}:
+
+    - [a1 :: l1] is smaller than [a2 :: l2] (negative result)
```

Comment:
The `(negative result)` reads weird here. I scratched my head. I would rather say: `The result is negative if ... is smaller than ..., that is`

Reply:
Thanks for the good feedback. I didn't find a nice formulation to at the same time explain lexicographic ordering (yes) and explain the integer convention, so I just dropped the second part -- hopefully the reference to `Stdlib.compare` is enough.